codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/AwsSdkGoV2ChangedNullability.java

```diff
@@ -0,0 +1,107 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.go.codegen.customization;
+
+import static software.amazon.smithy.aws.go.codegen.customization.ApiGatewayExportsNullabilityExceptionIntegration.getNullabilityExceptions;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import software.amazon.smithy.aws.traits.ServiceTrait;
+import software.amazon.smithy.diff.Differences;
+import software.amazon.smithy.diff.evaluators.AbstractDiffEvaluator;
+import software.amazon.smithy.go.codegen.knowledge.GoPointableIndex;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.knowledge.NullableIndex;
+import software.amazon.smithy.model.shapes.NumberShape;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.traits.DefaultTrait;
+import software.amazon.smithy.model.validation.ValidationEvent;
+
+/**
+ * An aws-sdk-go-v2 specific ChangedNullability diff evaluator.
+ *
+ * The following conditions will create events:
+ * - Added shapes with a default trait that are not in the nullability
+ *   exceptions.
+ *   - TODO(APIGW): remove after model fixes
+ * - Any changed shape that differs in nullability in GoPointableIndex and
+ *   NullableIndex
+ */
+public class AwsSdkGoV2ChangedNullability extends AbstractDiffEvaluator {
+    @Override
+    public List<ValidationEvent> evaluate(Differences differences) {
+        Model oldModel = differences.getOldModel();
+        GoPointableIndex oldGoPointableIndex = GoPointableIndex.of(oldModel);
+        NullableIndex oldNullableIndex = NullableIndex.of(oldModel);
+
+        Model newModel = differences.getNewModel();
+        GoPointableIndex newGoPointableIndex = GoPointableIndex.of(newModel);
+        NullableIndex newNullableIndex = NullableIndex.of(newModel);
+        Set<NumberShape> newNumberShapes = newModel.toSet(NumberShape.class);
+
+        List<ValidationEvent> events = new ArrayList<ValidationEvent>();
+
+        ShapeId service = newModel
+                .getServiceShapesWithTrait(ServiceTrait.class)
+                .iterator().next()
+                .toShapeId();
+        Set<ShapeId> nullabilityExceptions = getNullabilityExceptions(service);
+
+        differences.addedShapes().forEach(shape -> {
+            // TODO(APIGW): remove after model fixes
+            if (shape.hasTrait(DefaultTrait.class)
+                    && (shape.isBooleanShape() || newNumberShapes.contains(shape))
```

Comment:
I dont think this actually catches all of the cases we want. It doesnt check if there is a member shape marked with a default value that targets a number or boolean shape.

I think using selectors here might be more robust. I think the following two selectors cover all cases: 
```
member[trait|smithy.api#default]:test(> :is(boolean, number))
```
```
[trait|smithy.api#default]:is(boolean, number)
```